File: TOMIMSV4/Entity/Mantenimientos/Bodega/Bodega_Tramo/clsBeBodega_tramo.vb
<Serializable>
Public Class clsBeBodega_tramo
    Implements ICloneable
    Public Property IdTramo() As Integer = 0
    Public Property IdSector() As Integer = 0
    Public Property IdArea() As Integer = 0
    Public Property IdBodega() As Integer = 0
    Public Property Sistema() As Boolean = False
    Public Property Descripcion() As String = ""
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Activo() As Boolean = False
    Public Property Alto() As Double = 0.0
    Public Property Largo() As Double = 0.0
    Public Property Ancho() As Double = 0.0
    Public Property Margen_izquierdo() As Double = 0.0
    Public Property Margen_derecho() As Double = 0.0
    Public Property Margen_superior() As Double = 0.0
    Public Property Margen_inferior() As Double = 0.0
    Public Property Codigo() As String = ""
    Public Property Indice_x() As Integer = 0
    Public Property Orientacion() As Integer = 0
    Public Property IdTipoProductoDefault() As Integer = 0
    Public Property IdFontEnc As Integer = 0
    Public Property IdTipoRack As Integer = 0
    Public Property Es_Rack As Boolean = True
    Public Property Horizontal As Boolean = False
    Public Property Orden_Descendente As Boolean = False

    Sub New()
    End Sub

    Sub New(ByRef IdTramo As Integer,
            ByVal IdSector As Integer,
            ByVal sistema As Boolean,
            ByVal descripcion As String,
            ByVal user_agr As String,
            ByVal fec_agr As Date,
            ByVal user_mod As String,
            ByVal fec_mod As Date,
            ByVal activo As Boolean,
            ByVal alto As Double,
            ByVal largo As Double,
            ByVal ancho As Double,
            ByVal margen_izquierdo As Double,
            ByVal margen_derecho As Double,
            ByVal margen_superior As Double,
            ByVal margen_inferior As Double,
            ByVal Codigo As String,
            ByVal Indice_x As Integer,
            ByVal Orientacion As Integer,
            ByVal IdTipoProductoDefault As Integer,
            ByVal IdFontEnc As Integer,
            ByVal Horizontal As Boolean,
            ByVal Orden_Descendente As Boolean)

        Me.IdTramo = IdTramo
        Me.IdSector = IdSector
        Me.Sistema = sistema
        Me.Descripcion = descripcion
        Me.User_agr = user_agr
        Me.Fec_agr = fec_agr
        Me.User_mod = user_mod
        Me.Fec_mod = fec_mod
        Me.Activo = activo
        Me.Alto = alto
        Me.Largo = largo
        Me.Ancho = ancho
        Me.Margen_izquierdo = margen_izquierdo
        Me.Margen_derecho = margen_derecho
        Me.Margen_superior = margen_superior
        Me.Margen_inferior = margen_inferior
        Me.Codigo = Codigo
        Me.Indice_x = Indice_x
        Me.Orientacion = Orientacion
        Me.IdTipoProductoDefault = IdTipoProductoDefault
        Me.IdFontEnc = IdFontEnc
        Me.Horizontal = Horizontal
        Me.Orden_Descendente = Orden_Descendente
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Datos_Maestros/clsBeBodega_tramo.cs
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;


namespace WMS.EntityCore.Datos_Maestros
{
    public class clsBeBodega_tramo : ICloneable
    {
        [Column("IdTramo")]
        [DisplayName("IdTramo")]
        public int IdTramo { get; set; } = 0;

        [Column("IdSector")]
        [DisplayName("IdSector")]
        public int IdSector { get; set; } = 0;

        [Column("sistema")]
        [DisplayName("sistema")]
        public bool Sistema { get; set; } = false;

        [Column("descripcion")]
        [DisplayName("descripcion")]
        public string Descripcion { get; set; } = "";

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("alto")]
        [DisplayName("alto")]
        public double Alto { get; set; } = 0;

        [Column("largo")]
        [DisplayName("largo")]
        public double Largo { get; set; } = 0;

        [Column("ancho")]
        [DisplayName("ancho")]
        public double Ancho { get; set; } = 0;

        [Column("margen_izquierdo")]
        [DisplayName("margen_izquierdo")]
        public double Margen_izquierdo { get; set; } = 0;

        [Column("margen_derecho")]
        [DisplayName("margen_derecho")]
        public double Margen_derecho { get; set; } = 0;

        [Column("margen_superior")]
        [DisplayName("margen_superior")]
        public double Margen_superior { get; set; } = 0;

        [Column("margen_inferior")]
        [DisplayName("margen_inferior")]
        public double Margen_inferior { get; set; } = 0;

        [Column("Codigo")]
        [DisplayName("Codigo")]
        public string Codigo { get; set; } = "";

        [Column("Indice_x")]
        [DisplayName("Indice_x")]
        public int Indice_x { get; set; } = 0;

        [Column("Orientacion")]
        [DisplayName("Orientacion")]
        public int Orientacion { get; set; } = 0;

        [Column("IdTipoProductoDefault")]
        [DisplayName("IdTipoProductoDefault")]
        public int IdTipoProductoDefault { get; set; } = 0;

        [Column("IdFontEnc")]
        [DisplayName("IdFontEnc")]
        public int IdFontEnc { get; set; } = 0;

        [Column("IdTipoRack")]
        [DisplayName("IdTipoRack")]
        public int IdTipoRack { get; set; } = 0;

        [Column("es_rack")]
        [DisplayName("es_rack")]
        public bool Es_rack { get; set; } = false;

        [Column("Horizontal")]
        [DisplayName("Horizontal")]
        public bool Horizontal { get; set; } = false;

        [Column("IdArea")]
        [DisplayName("IdArea")]
        public int IdArea { get; set; } = 0;

        [Column("IdBodega")]
        [DisplayName("IdBodega")]
        public int IdBodega { get; set; } = 0;

        [Column("orden_descendente")]
        [DisplayName("orden_descendente")]
        public bool Orden_descendente { get; set; } = false;

        public clsBeBodega_tramo() { }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}
